llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -5811,6 +5811,11 @@ bool RISCVTargetLowering::isShuffleMaskLegal(ArrayRef<int> M, EVT VT) const {
   if (SVT.getScalarType() == MVT::i1)
     return false;
 
+  unsigned NumElts = M.size();
+  if (ShuffleVectorInst::isReverseMask(M, NumElts) &&
+      ShuffleVectorInst::isSingleSourceMask(M, NumElts))
```

Comment:
I think isReverseMask checks isSingleSourceMask.

Reply:
So it does.  I swear I checked that, but I guess not.  Will remove.